### PR TITLE
Expose some skip inspection options for `armi.init` and `db.loadOperator`

### PR DESCRIPTION
**Pull request by opotowsky**
## What is the change?

This exposes a `skipInspection` option to `armi.init` and a `handleInvalids` option `db.loadOperator`. The original behavior is retained by default.

## Why is the change being made?

Folks write scripts that use the API, and when they run these scripts, they are often stuck in settings inspection mode (typing `y` + `<enter>` over and over if they are loading older settings files). This helps them skip said inspections so they can run scripts without babysitting them.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code. (Test added for loadOperator, but ARMI init is untested.)

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Review by onufer — APPROVED** [review]

**Comment by john-science** [comment]
Looks great.  It really could use some unit testing for the DB stuff.  (Testing `armi.init()` can be skipped.)

**Comment by opotowsky** [comment]
> Is this only needed for interactive mode? What's the scenario where someone needs to babysit settings queries? Not a huge fan of adding more kwargs to init so I'm curious if we have the right check inputs design.

Yes this is needed for interactive mode, when someone runs a python script on the command line.

File: armi/__init__.py [hunk 1]
```diff
@@ -144,6 +144,9 @@ def init(choice=None, fName=None, cs=None):
         If supplied, this CS object will supercede the other case input methods and use
         the object directly.
 
+    skipInspection : bool, optional
+        Whether or not the inputs should be checked for valid settings. Default is True.
```
Comment (john-science) [inline armi/__init__.py:148]:
```suggestion
        Whether or not the inputs should be checked for valid settings. Default is False.
```

File: doc/release/0.4.rst [hunk 2]
```diff
@@ -27,6 +27,7 @@ New Features
 #. Improve efficiency of reaction rate calculations. (`PR#1887 <https://github.com/terrapower/armi/pull/1887>`_)
 #. Adding new options for simplifying 1D cross section modeling. (`PR#1949 <https://github.com/terrapower/armi/pull/1949>`_)
 #. Updating ``copyOrWarn`` and ``getFileSHA1Hash`` to support directories. (`PR#1984 <https://github.com/terrapower/armi/pull/1984>`_)
+#. Expose some skip inspection options for ``armi.init`` and ``db.loadOperator``. (`PR#XXXX <https://github.com/terrapower/armi/pull/1984>`_)
```
Comment (john-science) [inline doc/release/0.4.rst:30]:
```suggestion
#. Expose some skip inspection options for ``armi.init`` and ``db.loadOperator``. (`PR#2005 <https://github.com/terrapower/armi/pull/2005>`_)
```

File: armi/bookkeeping/db/__init__.py [hunk 3]
```diff
@@ -104,6 +105,8 @@ def loadOperator(
     allowMissing : bool
         Whether to emit a warning, rather than crash if reading a database
         with undefined parameters. Default False.
+    handleInvalids : bool
+        Whether to check for invlaid settings. Default True.
```
Comment (opotowsky) [inline armi/bookkeeping/db/__init__.py:109]:
```suggestion
        Whether to check for invalid settings. Default True.
```

File: armi/bookkeeping/db/database.py [hunk 4]
```diff
@@ -393,9 +393,14 @@ def fileName(self, fName):
...
         self._fileName = fName
 
-    def loadCS(self):
+    def loadCS(self, handleInvalids=True):
         """Attempt to load settings from the database file.
 
+        Parameters
+        ----------
+        handleInvalids : bool
+            Whether to check for invlaid settings. Default True.
```
Comment (opotowsky) [inline armi/bookkeeping/db/database.py:402]:
```suggestion
            Whether to check for invalid settings. Default True.
```

File: armi/bookkeeping/db/database.py [hunk 5]
```diff
@@ -743,6 +751,8 @@ def load(
         allowMissing : bool, optional
             Whether to emit a warning, rather than crash if reading a database
             with undefined parameters. Default False.
+        handleInvalids : bool
+            Whether to check for invlaid settings. Default True.
```
Comment (opotowsky) [inline armi/bookkeeping/db/database.py:755]:
```suggestion
            Whether to check for invalid settings. Default True.
```

File: armi/bookkeeping/db/__init__.py [hunk 6]
```diff
@@ -86,6 +86,7 @@ def loadOperator(
     loadNode,
     statePointName=None,
     allowMissing=False,
+    handleInvalids=True,
```
Comment (onufer) [inline armi/bookkeeping/db/__init__.py:89]: suggest rename since handle is unclear. "query invalids" or something like that. handel sounds like ignore.

Comment (opotowsky) [reply armi/bookkeeping/db/__init__.py:89]: this is already an arg name used in ARMI and I was just exposing it further back in the chain. Renaming would be a bigger effort.

File: armi/bookkeeping/db/tests/test_database3.py [hunk 7]
```diff
@@ -206,10 +208,6 @@ def setUp(self):
         self.db: database.Database = self.dbi.database
         self.stateRetainer = self.r.retainState().__enter__()
 
-        # used to test location-based history. see details below
-        self.centralAssemSerialNums = []
-        self.centralTopBlockSerialNums = []
-
```
Comment (opotowsky) [inline armi/bookkeeping/db/tests/test_database3.py:212]: this is just cleanup....unused code

File: armi/bookkeeping/db/tests/test_database3.py [hunk 8]
```diff
@@ -226,6 +224,37 @@ def makeHistory(self):
...
 
+    def test_loadOperator(self):
+        self.makeHistory()
+        self.db.close()
+        # Write a bad setting
+        with h5py.File(self.db.fileName, "r+") as hf:
+            settings = hf["inputs/settings"].asstr()[()]
+            settings += "  fakeSetting: I'll be back"
+            del hf["inputs/settings"]
+            hf["inputs/settings"] = settings
```
Comment (opotowsky) [inline armi/bookkeeping/db/tests/test_database3.py:235]: ok so I didn't have time to dig in on how to save new settings to the DB, but that's the intent here. Once that is done I bet the test will work!

File: armi/bookkeeping/db/tests/test_database3.py [hunk 9]
```diff
@@ -226,6 +224,37 @@ def makeHistory(self):
...
+        # Test with no complaints
+        with mockRunLogs.BufferLog() as mock:
+            o = loadOperator(
+                self._testMethodName + ".h5",
+                0,
+                0,
+                allowMissing=False,
```
Comment (opotowsky) [inline armi/bookkeeping/db/tests/test_database3.py:242]:
```suggestion
                allowMissing=True,
```

File: armi/bookkeeping/db/tests/test_database3.py [hunk 10]
```diff
@@ -226,6 +224,37 @@ def makeHistory(self):
...
+            )
+            self.assertNotIn("fakeSetting", mock.getStdout())
+            self.assertEqual(o.cs["fakeSetting"], "I'll be back")
+        # test with complaints
+        with mockRunLogs.BufferLog() as mock:
+            o = loadOperator(
+                self._testMethodName + ".h5",
+                0,
+                0,
+                allowMissing=False,
```
Comment (opotowsky) [inline armi/bookkeeping/db/tests/test_database3.py:253]:
```suggestion
                allowMissing=True,
```

File: armi/bookkeeping/db/tests/test_database3.py (hunk 8 above)
Comment (opotowsky) [reply armi/bookkeeping/db/tests/test_database3.py:235]: once this test works I might be able to add more

Comment (opotowsky) [reply armi/bookkeeping/db/tests/test_database3.py:235]: yay figured it out! pushing up new tests soon

File: doc/release/0.5.rst [hunk 11]
```diff
@@ -50,6 +50,7 @@ New Features
 #. Improve efficiency of reaction rate calculations. (`PR#1887 <https://github.com/terrapower/armi/pull/1887>`_)
 #. Adding new options for simplifying 1D cross section modeling. (`PR#1949 <https://github.com/terrapower/armi/pull/1949>`_)
 #. Adding ``--skip-inspection`` flag to ``CompareCases`` CLI. (`PR#1842 <https://github.com/terrapower/armi/pull/1842>`_)
+#. Expose some skip inspection options for ``armi.init`` and ``db.loadOperator``. (`PR#2005 <https://github.com/terrapower/armi/pull/2005>`_)
```
Comment (john-science) [inline doc/release/0.5.rst:53]:
```suggestion
#. Exposing skip inspection options for ``armi.init`` and ``db.loadOperator``. (`PR#2005 <https://github.com/terrapower/armi/pull/2005>`_)
```

Comment (john-science) [reply doc/release/0.5.rst:53]: Not interesting. We use active "ing" verbs to start our commits. Like 95% of the time.

I'm not in love with it, but it's a thing we do, so whatever.

**Review by john-science — APPROVED** [review]

**Pull request merged**